### ISSUE TITLE
App crashes on Android

### ISSUE THREAD

**Issue by flonghini** [sparkfabrik/sparkfabrik-react-native-idfa-aaid#5]
Hi there, congrats on this lib it's been hard to find nowadays one that actually works.

 I've followed the README instructions and on iOS it works, it successfully builds on Android but when I run the App it crashes, no errors are displayed. 

Please is there any additional configuration that is needed to run on Android? Is there any additional information that I can provide?

**Comment by flonghini** [comment]
Hi again, I was able to run on Android after doing this:

1. Update your AndroidManifest.xml and declare that your app is an Ad Manager app, as instructed on Google's Ad Manager guide:

`<manifest>
    <application>
        <meta-data
            android:name="com.google.android.gms.ads.AD_MANAGER_APP"
            android:value="true"/>
    </application>
</manifest>`

2. Add on android/settings.gradle :
`include ':sparkfabrikreactnativeidfaaaid'
project(':sparkfabrikreactnativeidfaaaid').projectDir = new File(rootProject.projectDir, '../node_modules/@sparkfabrik/react-native-idfa-aaid/android')`

**Comment by edodusi** [comment]
Hi @flonghini, thanks for reporting this, can you give me more context before I try to reproduce and fix that?

Which version of react native are you running, which library are you using for admanager or admob, and does the crash happen on the android emulator or real device? 

Also if you run the emulator, can you get the runtime logs from android studio?

Thanks

**Comment by flonghini** [comment]
Hi @edodus, thanks for your quick reply.

I'm using your lib to retrieve IDFA/AAID for integration with Appsflyer (react-native-appsflyer": "^6.1.10"). The app crashes on an emulator and device (Pixel 3 / Moto E4) without any error logs on Android Studio, successfully builds, and when the intent is launched it just crashes silently. 

In order to isolate any additional lib, I've made another test creating a branding new project with create-react-native-app and you could also try it. What I have done:

1. create-react-native-app my-project
2. yarn add @sparkfabrik/react-native-idfa-aaid
3. run the project

On step 3 it crashed, then I added the manual configs on AndroidManifest.xml and on settings.graddle, then I've re-runned the project and it works.

thank you

**Comment by edodusi** [comment]
@flonghini I think I found out why this is happening and the way to fix it. Seems related to the fact that the `ads` library from `play-services` is a requirement on current release, because we took for granted that the host project was already coonfigured as a google-ads project.
But this is a mistake, so now I changed the dependency library from `ads` to `aaid` (which apparently only gives you this interface) and testing with a blank app I was able to ged the aaid without crashes and without having an APP_ID in the Manifest.

We will release a new version with this fix so hopefully in a few hours you will find an update to test this out.

Waiting for your feedback on the updated module!
